connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -540,10 +601,13 @@ Map<String, Config> describeTopicConfigs(Set<String> topics)
 
     Config targetConfig(Config sourceConfig) {
         List<ConfigEntry> entries = sourceConfig.entries().stream()
-            .filter(x -> !x.isDefault() && !x.isReadOnly() && !x.isSensitive())
-            .filter(x -> x.source() != ConfigEntry.ConfigSource.STATIC_BROKER_CONFIG)
-            .filter(x -> shouldReplicateTopicConfigurationProperty(x.name()))
-            .collect(Collectors.toList());
+                    .filter(x -> (!useIncrementalAlterConfigs.equals(MirrorSourceConfig.NEVER_USE_INCREMENTAL_ALTER_CONFIG)
```

Comment:
Isn't it possible that `useIncrementalAlterConfigs` changes between when we read it here and when we read it in `updateTopicConfigs`?

Reply:
Also, doesn't this result in a change in behavior for new topic creation (where we also use this method)?

Reply:
I think we may want to alter this method to take a boolean `incremental` (rename as you like) parameter and use that instead of checking the value of `useIncrementalAlterConfigs`.

That way, when creating new topics, we can hardcode that parameter to `false`, and when syncing topics, we can check the value of `useIncrementalAlterConfigs` once in the beginning, and then use that single check to determine whether we're going to use the  incremental or deprecated API.

Reply:
This is looking better, but not quite--we still aren't handling the possibility that `useIncrementalAlterConfigs` changes in the middle of an invocation of `syncTopicConfigs`. (Left a more detailed comment in the review.)